File: Leetcode/421. Maximum XOR of Two Numbers in an Array/maximum_XOR_of_two_numbers_in_an_array.py
class Solution:
    def findMaximumXOR(self, nums) -> int:
        ans = 0
        for i in range(32)[::-1]:
            ans <<= 1
            prefix = {num>>i for num in nums}
            ans += any(ans^1^a in prefix for a in prefix)
        return ans
    

sol = Solution()
print(sol.findMaximumXOR([14,70,53,83,49,91,36,80,92,51,66,70]))

File: Leetcode/421. Maximum XOR of Two Numbers in an Array/maximum_XOR_of_two_numbers_in_an_array.rs
use std::collections::HashSet;

struct Solution;

impl Solution {
    pub fn find_maximum_xor(nums: Vec<i32>) -> i32 {
        let mut ans = 0;
        for i in (0..32).rev() {
            ans <<= 1;
            let prefixes: HashSet<i32> = nums.iter().map(|&num| num >> i).collect();
            ans += prefixes.iter().any(|&a| prefixes.contains(&(ans ^ 1 ^ a))) as i32;
        }
        ans
    }
}

fn main() {
    let sol = Solution::find_maximum_xor(vec![14, 70, 53, 83, 49, 91, 36, 80, 92, 51, 66, 70]);
    println!("{:?}", sol);
}